yarn-project/aztec.js/src/contract/base_contract_interaction.ts

```diff
@@ -67,12 +62,19 @@ export abstract class BaseContractInteraction {
    * the AztecAddress of the sender. If not provided, the default address is used.
    * @returns A SentTx instance for tracking the transaction status and information.
    */
-  public send(options: SendMethodOptions = {}) {
+  public send(options: SendMethodOptions = {}): SentTx {
     const promise = (async () => {
-      const tx = this.tx ?? (await this.prove(options));
-      return this.wallet.sendTx(tx);
+      // Don't call this.prove() to avoid the serialization of the ProvenTx object.
```

Comment:
Is it expensive? If not, I'd prefer creating an unneeded object over duplicating code.

Reply:
Not really expensive, issue is having to implement `.toJSON()` for the whole wallet object, since it would move back and forth. Might be misunderstanding how the RPC works, but I had trouble when I tried

Reply:
Created a protected method `proveInternal` to avoid duplication. Not super stoked on the name, but can be improved in pt. 2.

Reply:
I alternate between calling these kind of methods `methodInternal` and `doMethod`. I don't like either.